File: src/main/scala/App/HbaseTest.scala
package main.scala.App

import org.apache.hadoop.hbase.client.{ConnectionFactory, Get, Table}
import org.apache.hadoop.hbase.util.Bytes
import org.apache.hadoop.hbase.{HBaseConfiguration, TableName}

/**
  * Created by Administrator on 2016/8/12.
  */
object HbaseTest {


  def readData(table: Table): Unit = {
    //查询某条数据
    val g = new Get("83540c57b01248779b23ae055a4182df".getBytes)
    val result = table.get(g)
    val SERVICEID = Bytes.toString(result.getValue("SERVICE_INFO".getBytes, "SERVICEID".getBytes))
    val SERVICE_NAME = Bytes.toString(result.getValue("SERVICE_INFO".getBytes, "SERVICE_NAME".getBytes))
    println("GET id>>> :" + SERVICEID+">>>>>>>>>>>>>"+SERVICE_NAME)

  }

  def main(args: Array[String]): Unit = {
    System.setProperty("hadoop.home.dir", "D:\\Program Files\\Java\\hadoop-2.6.2");

    //val sc = new SparkContext(args(0), "HBaseTest",
    //System.getenv("SPARK_HOME"), SparkContext.jarOfClass(this.getClass))
    //val sc = new SparkContext("local","SparkHbase");

    val conf = HBaseConfiguration.create();
    conf.set("hbase.zookeeper.property.clientPort", "2181");
    conf.set("hbase.zookeeper.quorum", "nn1.hadoop.com");

    val conn = ConnectionFactory.createConnection(conf);
    val admin = conn.getAdmin;
    val userTable = TableName.valueOf("H_SERVICE_INFO")



    try {
      //获取 user 表
      val table = conn.getTable(userTable)

      try {

        //查询一条数据
        readData(table);

      } finally {
        if (table != null) table.close()
      }

    } finally {
      conn.close()
    }
  }



}

File: src/main/java/HbaseTest.java
package main.java;

import org.apache.hadoop.conf.Configuration;
import org.apache.hadoop.hbase.Cell;
import org.apache.hadoop.hbase.CellUtil;
import org.apache.hadoop.hbase.HBaseConfiguration;
import org.apache.hadoop.hbase.client.Get;
import org.apache.hadoop.hbase.client.HTable;
import org.apache.hadoop.hbase.client.Result;
import org.apache.hadoop.hbase.util.Bytes;

import java.io.IOException;
import java.io.Serializable;

/**
 * Created by Administrator on 2016/8/11.
 */
public class HbaseTest implements Serializable {
    public static HTable getTable(String tableName) throws IOException {
        // 创建HBase的配置对象
        Configuration conf = HBaseConfiguration.create();

        // 获取HBase的Table
        HTable table = new HTable(conf, "H_SERVICE_INFO");

        return table;

    }


    /**
     * 获取HBase Table数据
     *
     * @throws IOException
     */
    public static void getData() throws IOException {
    // 获取HBase Table
        HTable table = getTable("H_SERVICE_INFO");

    // 获取RowKey
        Get get = new Get(Bytes.toBytes("83540c57b01248779b23ae055a4182df"));
    // 根据列名称获取指定列
        // get.addColumn(Bytes.toBytes("info"), Bytes.toBytes("name")) ;

        Result result = table.get(get);
        Cell[] cells = result.rawCells();

        for (Cell cell : cells) {
            System.out.print(Bytes.toString(CellUtil.cloneFamily(cell)) + ":");
            System.out.print(Bytes.toString(CellUtil.cloneQualifier(cell))
                    + "-->");
            System.out.println(Bytes.toString(CellUtil.cloneValue(cell)));
        }

        table.close();
    }



    public static void main(String[] args) throws IOException {
    System.setProperty("hadoop.home.dir" , "D:\\Program Files\\Java\\hadoop-2.6.2" );
        getData();
    }
}
